### PR TITLE
Fix #23 - add `woab::scheduled_outside`

### LINKED ISSUE #23: Signals that get triggered synchronously
(continuing the discussion from #15 and https://github.com/idanarye/woab/pull/20#pullrequestreview-627452843)

Previously WoAB handled signals completely asynchronously - the signal handler would send a message to the Actix actor and would not wait for a result. If the handler needed to return a special value (inhibit decision) it would have to be configured by other means.

#20 changes it so that signal handlers would `block_on` the message, waiting for Actix to finish handling it, and have the result of that message be the inhibit decision if the signal requires one.

This approach raises a problem - some signals can be triggered from inside GTK functions. For example, if you are removing a widget from inside the Actix runtime, the `remove` signal will be triggered immediately - still inside the Actix runtime. This means you can't use Actix' `block_on`!

The current solution is to detect this scenario and when it happen - schedule the signal to run later and return `None` immediately. When the actual signal happens, it makes sure it didn't fail and that it returned `None` (to enforce consistency). This works for `remove`, which does not expect an inhibit decision, but I cannot guarantee it'll work properly for all signals. I experimented a bit and the very generic `event` signal - which does expect an inhibit decision - does run synchronously sometimes, but I'm not sure if that signal is actually used and is an actual problem.

Still - this is something that needs to be addressed.

### PR DESCRIPTION
**Pull request by idanarye**

### CODE REVIEW

File: examples/example_taggged_signals.rs [hunk 1]
```diff
@@ -80,9 +83,11 @@ impl actix::Handler<woab::Signal<usize>> for WindowActor {
                 None
             }
             "remove_addend" => {
-                if let Some((widgets, _)) = self.addends.remove(msg.tag()) {
-                    self.widgets.lst_addition.remove(&widgets);
-                    self.recalculate();
+                if let Some((addend, _)) = self.addends.remove(msg.tag()) {
+                    let lst_addition = self.widgets.lst_addition.clone();
+                    woab::schedule_outside(move || {
```
Comment (idanarye) [inline examples/example_taggged_signals.rs:88]: @piegamesde - this is the style. What do you think?

I don't love having to `clone()` things that need to be sent outside with the closure, but at least this is a burden Rust coders should be familiar with and know how to handle. And I don't really see a way around it.

Comment (piegamesde) [reply examples/example_taggged_signals.rs:88]: Cloning is a pattern so common to plain Rust gtk, that there is a convenient macro for it. I didn't miss it after switching to WoAB (although I still used once or twice), but I don't really mind it either.

File: src/signal_routing.rs [hunk 2]
```diff
@@ -209,28 +197,16 @@ impl<T: Clone + 'static> crate::GenerateRoutingGtkHandler for (T, NamespacedSign
...
-                            let signal_name = signal_name.as_ref().map(|s| s.as_str()).unwrap_or("???");
-                            panic!("Expected {:?} handler to return None - got {:?}", signal_name, result);
-                        }
-                    });
-                    None
-                }
+            let result = if let Ok(result) = crate::try_block_on(target.recipient.send(signal)) {
+                result.unwrap().unwrap()
+            } else {
+                panic!("Signal {:?} triggered from inside the Actix runtime. Try running whatever triggered it with `woab::schedule_outside()`", signal_name)
```
Comment (piegamesde) [inline src/signal_routing.rs:203]: Put user provided strings in quotes
```suggestion
                panic!("Signal '{:?}' triggered from inside the Actix runtime. Try running whatever triggered it with `woab::schedule_outside()`", signal_name)
```

Comment (idanarye) [reply src/signal_routing.rs:203]: No need to - the `{:?}` is already putting it in quotes. Also escapes special characters (not that I expect any, but you never know...)

Comment (piegamesde) [reply src/signal_routing.rs:203]: Uh, that's fancy, didn't know that.

**Pull request merged**